crates/rtmp/src/command_messages/netstream/reader.rs

```diff
@@ -0,0 +1,237 @@
+                    unreachable!();
+                };
+                let publishing_type = From::<&str>::from(&publishing_type);
+
+                if let NetStreamCommandPublishPublishingType::Unknown(publishing_type) = &publishing_type {
+                    tracing::warn!(publishing_type = ?publishing_type, "unknown publishing type in publish command");
+                }
+
+                Ok(Some(Self::Publish {
+                    publishing_name,
+                    publishing_type,
+                }))
+            }
+            "seek" => Ok(Some(Self::Seek)),
+            "pause" => Ok(Some(Self::Pause)),
+            "onStatus" => {
+                // skip command object
+                decoder.decode_with_type(Amf0Marker::Null)?;
+
+                let Amf0Value::Object(info_object) = decoder.decode_with_type(Amf0Marker::Object)? else {
+                    unreachable!();
+                };
+                // we have to get ownership here because we have to own the inner Cows
+                let mut info_object = info_object.into_owned();
+
+                let tc_url = if let Some(idx) = info_object.iter().position(|(k, _)| k == "tcUrl") {
+                    let (_, Amf0Value::String(tc_url)) = info_object.remove(idx) else {
+                        return Err(CommandError::InvalidOnStatusInfoObject);
+                    };
+                    Some(tc_url)
+                } else {
+                    None
+                };
+
+                let (_, Amf0Value::String(level)) = info_object
+                    .iter()
+                    .find(|(k, _)| k == "level")
+                    .ok_or(CommandError::InvalidOnStatusInfoObject)?
+                else {
+                    return Err(CommandError::InvalidOnStatusInfoObject);
+                };
+
+                let level = From::<&str>::from(level);
+
+                let (_, Amf0Value::String(code)) = info_object.remove(
+                    info_object
+                        .iter()
+                        .position(|(k, _)| k == "code")
+                        .ok_or(CommandError::InvalidOnStatusInfoObject)?,
+                ) else {
+                    return Err(CommandError::InvalidOnStatusInfoObject);
+                };
+
+                let description = if let Some(idx) = info_object.iter().position(|(k, _)| k == "description") {
+                    let (_, Amf0Value::String(description)) = info_object.remove(idx) else {
+                        return Err(CommandError::InvalidOnStatusInfoObject);
+                    };
+                    Some(description)
+                } else {
+                    None
+                };
+
+                Ok(Some(Self::OnStatus {
+                    tc_url,
+                    level,
+                    code,
+                    description,
+                }))
+            }
+            _ => Ok(None),
```

Comment:
maybe we should consider supporting the other commands ffmpeg sends and just returning to the user. instead of warning